lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/EvaluatorSettings.cs

```diff
@@ -89,7 +96,22 @@ internal sealed class EvaluatorSettings
                 Utilities.Diagnostics.Exceptions.Throw(
                     new ArgumentException("empty or null rootContextConfigString"), Logger);
             }
-            _rootContextConfig = new ContextConfiguration(rootContextConfigString);
+            _rootContextConfig = _serializer.FromString(rootContextConfigString);
+
+            try
+            {
+                _rootContextId = injector.ForkInjector(_rootContextConfig).GetNamedInstance<ContextConfigurationOptions.ContextIdentifier, string>();
```

Comment:
That is fine, but this is just a stop gap. All of this will be removed and `ContextConfiguration` won't even be needed at all. All the information will be injected directly into `ContextLifeCycle` instead,